### PR TITLE
feat: add filterable word list to content section

### PR DESCRIPTION
**Pull request by SiTaggart**
**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: packages/paste-website/src/components/tokens-list/NoTokensFound.tsx [hunk 1]
```diff
@@ -29,8 +29,8 @@ export const NoTokensFound: React.FC<NoTokensFoundProps> = ({onClearSearch}) =>
         </Heading>
         <Stack orientation="vertical" spacing="space70">
           <Text as="span">
-            Try adjusting your criteria to filter by the token name or value. If you still can’t find what you’re
-            looking for, reach out by creating a{' '}
+            Try adjusting your criteria to filter by the token name or value. If you still can&rsquo;t find what
+            you&rsquo;re looking for, reach out by creating a{' '}
```
[the remark this answers is not in the document]
Comment (SiTaggart) [reply packages/paste-website/src/components/tokens-list/NoTokensFound.tsx:33]: Generally speaking it's best practice to use entity encoded character. Used to be mainly around internationalization where different languages wouldn't necessarily support the character, so encoding them ensures they render correctly. We even have an eslint rule for it, but I think it's set to warn, not error.

File: packages/paste-website/src/components/word-list/EmptyState.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+interface EmptyStateProps {
+  onClearSearch: () => void;
+}
+
+export const EmptyState: React.FC<EmptyStateProps> = ({onClearSearch}) => (
+  <Card data-cy="word-list-empty-state" padding="space150">
+    <Box display="flex" columnGap="space110">
+      <Box flexShrink={0} size="size20">
+        <StaticImage
+          src="../../assets/images/tokens/no-tokens-found.svg"
```
[the remark this answers is not in the document]
Comment (SiTaggart) [reply packages/paste-website/src/components/word-list/EmptyState.tsx:20]: Yeah, I think you maybe right

File: packages/paste-website/src/components/word-list/WordFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+type WordFilterProps = {
+  value: string;
+  onInput: (e: React.FormEvent<HTMLInputElement>) => void;
+  onClear: () => void;
+};
+export const WordFilter: React.FC<WordFilterProps> = ({value, onInput, onClear}) => {
+  const filterID = useUID();
+  return (
+    <Box marginBottom="space70" maxWidth={['100%', '100%', 'size30']}>
+      <Label htmlFor={filterID}>Filter words:</Label>
```
Comment (gloriliale) [inline packages/paste-website/src/components/word-list/WordFilter.tsx:19]: nitpick: I think our guidance says to not put colons at the end of labels (though I can't find anything saying that now).

**Review by TheSisb — DISMISSED** [review]
This is soooo goood!

File: packages/paste-website/src/pages/foundations/content/word-list/index.mdx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+title: Words list
+description: List of visual attributes (background colors, spacing, etc.) that our design language is based on.
+slug: /foundations/content/word-list/
+---
+
+import {Box} from '@twilio-paste/box';
+import {SidebarCategoryRoutes} from '../../../../constants';
+import {WordsList} from '../../../../components/word-list';
+
+# Word List
```
Comment (gloriliale) [inline packages/paste-website/src/pages/foundations/content/word-list/index.mdx:11]: suggestion: it'd be better to use <GenericHeader> here, for consistency

**Review by gloriliale — DISMISSED** [review]
Left two suggestions, both pretty minor. Loving this word list!

**Review by shleewhite — APPROVED** [review]

File: packages/paste-website/src/components/word-list/utils.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import type {WordListShape} from './types';
+
+type FilterWordListArgs = (wordList: WordListShape, filterString: string) => Partial<WordListShape>;
+
+export const filterWordList: FilterWordListArgs = (wordList, filterString) => {
```
[the remark this answers is not in the document]
Comment (SiTaggart) [reply packages/paste-website/src/components/word-list/utils.tsx:5]: Done

File: cypress/integration/word-list/index.spec.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
+describe('Word list with no filters', function () {
+  beforeEach(() => {
+    cy.visit('/foundations/content/word-list');
+  });
+  it('should have a pre-populated list of words in the table', () => {
+    cy.get('[data-cy="word-list-table"] tbody tr').its('length').should('eq', 81);
```
[the remark this answers is not in the document]
Comment (SiTaggart) [reply cypress/integration/word-list/index.spec.ts]: done

File: packages/paste-website/src/pages/foundations/content/word-list/index.mdx (hunk 4 above)
Comment (SiTaggart) [reply packages/paste-website/src/pages/foundations/content/word-list/index.mdx]: done

File: packages/paste-website/src/components/word-list/WordFilter.tsx (hunk 3 above)
Comment (SiTaggart) [reply packages/paste-website/src/components/word-list/WordFilter.tsx]: done

**Review by gloriliale — APPROVED** [review]

**Review by kelseycavitt — APPROVED** [review]
Yay, this is so simple, but will be impactful quick!

**Review by shleewhite — APPROVED** [review]

**Pull request merged**